### PR TITLE
fix: simply nav bar on public page

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/5448

Screenshot from 2020-11-02 18-41-17

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mariobehling** [comment]
Very nice. I missed this: For not-logged in users we also need to take out "pricing", please on the event pages only.

Screenshot from 2020-11-02 14-34-36

File: app/components/nav-bar.js [hunk 1]
```diff
@@ -1,9 +1,14 @@
 import classic from 'ember-classic-decorator';
-import { action } from '@ember/object';
+import { action, computed } from '@ember/object';
 import Component from '@ember/component';
 
 @classic
 export default class NavBar extends Component {
+  @computed('session.currentRouteName')
+  get currentRoute() {
```
Comment (iamareebjamal) [inline app/components/nav-bar.js:8]: This doesn't return the current route, it returns if it is public route. Change to `isEventPageRoute`

File: app/templates/components/nav-bar.hbs [hunk 2]
```diff
@@ -10,9 +10,11 @@
         @route="pricing" class="item"
         @activeClass="">{{t 'Pricing'}}</LinkTo>
     {{/if}}
+    {{#if currentRoute}}
```
Comment (iamareebjamal) [inline app/templates/components/nav-bar.hbs:13]: `this.` is missing

Will throw a warning

File: app/templates/components/nav-bar.hbs [hunk 3]
```diff
@@ -5,14 +5,16 @@
       <h3>{{this.settings.appName}}</h3>
     </LinkTo>
     <div class="right menu nav-bar">
-    {{#if (not this.session.isAuthenticated)}}
+    {{#if (and (not this.session.isAuthenticated) this.isEventPageRoute)}}
```
Comment (iamareebjamal) [inline app/templates/components/nav-bar.hbs:8]: This means show Pricing if user is not logged in and the page is event page

File: app/components/nav-bar.js [hunk 4]
```diff
@@ -1,9 +1,14 @@
 import classic from 'ember-classic-decorator';
-import { action } from '@ember/object';
+import { action, computed } from '@ember/object';
 import Component from '@ember/component';
 
 @classic
 export default class NavBar extends Component {
+  @computed('session.currentRouteName')
+  get isEventPageRoute() {
```
Comment (iamareebjamal) [inline app/components/nav-bar.js:8]: This returns true if the page is not event page. It should either be named `isNotEventPageRoute` or return opposite of what it does. Please check the function names make sense. Change the name

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**